### ISSUE TITLE
[BUG] CSS-declared `border-radius` stops working on `AnimateSharedLayout` change until page refresh in `3.6.2`.

### ISSUE THREAD

**Issue by ivanburlakov** [framer/motion#996]
Sandbox: https://codesandbox.io/s/framer-motion-2-animatesharedlayout-animate-between-different-components-forked-te1yv

+ text-declared colors, like `red` stoped animating (with console warning), but maybe that is intentional. If not - can open separate issue.

**Comment by DanProudfoot** [comment]
I'm having this issue as well. Animating with AnimateSharedLayout using a kind of card layout. The card has border-radius, click the card to bring up full preview, and on reset the card has lost it's border-radius. The border-radius is forced to be 0% at this point

**Comment by msobie** [comment]
Experienced the same issue since upgrading to v3.6.2. It appears that motion._element_ is overriding custom css classes after the initial animation. If you inspect the element, you will see element.style with `border-radius: 0%;` added, which disables any custom border-radius.

I also have noticed my AnimateSharedLayout slider get vertically offset on scroll. Upon inspection, it appears to add `transform: translate3d(0px, 13px, 0px) scale(1, 1);`. From what I can gather, it happens if you scroll soon after triggering the animation, and is only applied for a single point on the scroll axis. If you keep scrolling, it resets immediately. However, if you scroll slowly, you can get the style to stick (see screenshot below). I can open a separate issue for this if needed, but I have a feeling these two issues are related.

![image](https://user-images.githubusercontent.com/70611133/108611688-dc275100-73ae-11eb-82c4-72fb415b8809.png)

**Comment by mattgperry** [comment]
@msobie can you confirm the second issue doesn’t happen with 3.2.0? And file a new issue with a sandbox

**Comment by mattgperry** [comment]
Thanks for the report! I'm going to take a look at this tomorrow. As for the named colors, Motion never did/will support animating named colours. I'm surprised if you encountered an error where the previously wasn't one, but it isn't intended to be supported.

**Comment by ivanburlakov** [comment]
> As for the named colors, Motion never did/will support animating named colours. I'm surprised if you encountered an error where the previously wasn't one, but it isn't intended to be supported.

Yeah, my project was using `AnimateSharedLayout` between `red` and `green`, etc. for the long time before recent upgrading.

**Comment by mattgperry** [comment]
Ah I see. I suspect it was animating between colors by using opacity so it
looked like it was animating colors but wasn’t. In the new version it does
actually crossfade between backgroundColor hence the error. But it
shouldn’t be triggering on components that were previously animating
opacity.

**Comment by msobie** [comment]
@mattgperry After additional testing, I realized the second issue was the result of an onScroll event listener re-rendering the page on every scroll. The vertical offset I reported above was a result of poor implementation on my part. Apologies for mentioning it without further examination. Appreciate all your work, and thanks for patching the border-radius bug so fast!

**Comment by mattgperry** [comment]
That’s actually quite informative still thanks for the extra info

**Comment by msobie** [comment]
I put together a sandbox for you to take a look at. It only happens if you scroll immediately after triggering the animation. I typically found it easier to reproduce by using my mouse's scroll wheel, since you can guarantee a scroll while the animation is still firing.

Sandbox: https://codesandbox.io/s/framer-motion-issue-996-r23bp

**Comment by mattgperry** [comment]
Thanks, really appreciate it

**Comment by ShannonLCapper** [comment]
Is this issue related to https://github.com/framer/motion/issues/940 (and specifically, https://github.com/framer/motion/issues/940#issuecomment-847427853)?
